mmds/src/lib.rs

```diff
@@ -56,28 +56,14 @@ pub fn parse_request(request_bytes: &[u8]) -> Response {
                         Body::new(response_body),
                     )
                 }
-                Err(e) => {
-                    match e {
-                        MmdsError::NotFound => {
-                            // NotFound
-                            let error_msg = format!("Resource not found: {}.", uri);
-                            build_response(
-                                request.http_version(),
-                                StatusCode::NotFound,
-                                Body::new(error_msg),
-                            )
-                        }
-                        MmdsError::UnsupportedValueType => {
-                            // InternalServerError
-                            let error_msg =
-                                format!("The resource {} has an invalid format.", uri.to_string());
-                            build_response(
-                                request.http_version(),
-                                StatusCode::InternalServerError,
-                                Body::new(error_msg),
-                            )
-                        }
-                    }
+                Err(()) => {
```

Comment:
I believe for parsing the request you can still get `UnsupportedValueType`. So this code should remain unchanged.